webapp/tests/Unit/Controller/API/AccessControllerTest.php

```diff
@@ -0,0 +1,163 @@
+<?php declare(strict_types=1);
+
+namespace App\Tests\Unit\Controller\API;
+
+class AccessControllerTest extends BaseTest
+{
+    public function testAccessAsDemo(): void
+    {
+        $url = $this->helperGetEndpointURL('access');
+        $this->verifyApiJsonResponse('GET', $url, 403, 'demo');
+    }
+
+    public function testAccessAsAdmin(): void
+    {
+        $url    = $this->helperGetEndpointURL('access');
+        $access = $this->verifyApiJsonResponse('GET', $url, 200, 'admin');
+        self::assertArrayHasKey('capabilities', $access);
+        self::assertSame(
+            ['contest_start', 'team_submit', 'team_clar', 'proxy_submit', 'proxy_clar', 'admin_submit', 'admin_clar'],
+            $access['capabilities']
+        );
+
+        self::assertArrayHasKey('endpoints', $access);
+
+        $expectedEndpoints = [
```

Comment:
Is there really value in having the same hardcoded list here again?

Reply:
I’m not sure. I can check only for the endpoints maybe?